src/KOKKOS/fix_rigid_kokkos.cpp

```diff
@@ -0,0 +1,1835 @@
+  //
+  // also calls set_v, which modifies the following:
+  // v, atom_vec->omega, atom_vec->angmom
+  // uses the following:
+  // e{x,y,z}_space, displace, omega, v, vcm, atom_vec->mass, xcmimage, x
+
+  k_sum.sync<LMPHostType>();
+  k_all.sync<LMPHostType>();
+  k_torque.sync<LMPHostType>();
+  k_langextra.sync<LMPHostType>();
+  k_omega.sync<LMPHostType>();
+
+  atomKK->k_x.sync<LMPHostType>();
+  atomKK->k_v.sync<LMPHostType>();
+  atomKK->k_f.sync<LMPHostType>();
+
+  k_body.sync<LMPHostType>();
+  k_xcm.sync<LMPHostType>();
+  k_ex_space.sync<LMPHostType>();
+  k_ey_space.sync<LMPHostType>();
+  k_ez_space.sync<LMPHostType>();
+  k_xcmimage.sync<LMPHostType>();
+  k_inertia.sync<LMPHostType>();
+
+  k_vcm.sync<LMPHostType>();
+  k_quat.sync<LMPHostType>();
+
+  // modifies:
+  k_sum.modify<LMPHostType>();
+  k_all.modify<LMPHostType>();
+  k_torque.modify<LMPHostType>();
+  k_langextra.modify<LMPHostType>();
+  k_omega.modify<LMPHostType>();
+  k_quat.modify<LMPHostType>();
+  atomKK->k_v.modify<LMPHostType>();
+
+  FixRigid::setup(vflag);
+
+  k_sum.sync<DeviceType>();
+  k_all.sync<DeviceType>();
+  k_torque.sync<DeviceType>();
+  k_langextra.sync<DeviceType>();
+  k_omega.sync<DeviceType>();
+  k_quat.sync<DeviceType>();
+  atomKK->k_v.sync<DeviceType>();
+
+}
+
+
+template <class DeviceType>
+void FixRigidKokkos<DeviceType>::pre_neighbor()
+{
+  // pre_neighbor modifies both xcm and imagebody
+  // image_shift modifies xcmimage and body
+  k_xcm.sync<DeviceType>();
+  k_imagebody.sync<DeviceType>();
+
+  {
+	  // Local block for Kokkos lambda.
+    auto l_xcm = k_xcm.d_view;
+    auto l_imagebody = k_imagebody.d_view;
+    auto domainKK = static_cast<DomainKokkos *>(domain);
+
+    int periodic_bits = 0;
+    periodic_bits +=   domain->xperiodic;
+    periodic_bits += 2*domain->yperiodic;
+    periodic_bits += 4*domain->zperiodic;
+
+
+    Few<double,3> prd{domain->prd[0], domain->prd[1], domain->prd[2]};
+    Few<double,3> prd_lambda{domain->prd_lamda[0], domain->prd_lamda[1], domain->prd_lamda[2]};
+
+    Few<double,6> h{domain->h[0], domain->h[1], domain->h[2],
+                    domain->h[3], domain->h[4], domain->h[5]};
+    Few<double,6> h_inv{domain->h_inv[0], domain->h_inv[1], domain->h_inv[2],
+                        domain->h_inv[3], domain->h_inv[4], domain->h_inv[5]};
+
+
+    Few<double,3> boxlo{domain->boxlo[0], domain->boxlo[1], domain->boxlo[2]};
+    Few<double,3> boxhi{domain->boxhi[0], domain->boxhi[1], domain->boxhi[2]};
+
+    Few<double,3> boxlo_lambda{domain->boxlo_lamda[0], domain->boxlo_lamda[1], domain->boxlo_lamda[2]};
+    Few<double,3> boxhi_lambda{domain->boxhi_lamda[0], domain->boxhi_lamda[1], domain->boxhi_lamda[2]};
+
+    Kokkos::parallel_for(nbody, LAMMPS_LAMBDA(const int &ibody){
+      Few<double,3> xcm_ibody{l_xcm(ibody,0), l_xcm(ibody,1), l_xcm(ibody,2)};
+      imageint imagebody_ibody = l_imagebody(ibody);
+
+      auto new_xcm = domainKK->remap(prd, h, h_inv, triclinic, boxlo, boxhi,
```

Comment:
This will crash on a GPU because the pointer lives on the host CPU. Must use a static function call or something that can be captured by the lambda, for example see https://github.com/lammps/lammps/blob/master/src/KOKKOS/fix_momentum_kokkos.cpp#L169.